examples/models/image_classification/CNN.py

```diff
@@ -0,0 +1,201 @@
+                                            min_lr=0.00001)
+        history = self._model.fit_generator(datagen.flow(X_train, y_train, batch_size=bs),
+                              epochs=ep, validation_data = (X_val,y_val),
+                              verbose = 2, steps_per_epoch=X_train.shape[0] // bs,
+                              callbacks=[learning_rate_reduction]) 
+        
+        # Compute train accuracy
+        (train_loss, train_acc) = self._model.evaluate(X_val, y_val)
+        logger.log('Train loss: {}'.format(train_loss))
+        logger.log('Train accuracy: {}'.format(train_acc))
+
+
+    def evaluate (self, dataset_path):
+        dataset = dataset_utils.load_dataset_of_image_files(dataset_path, image_size=[28,28])
+        (images, classes) = zip(*[(image, image_class) for (image, image_class) in dataset])
+        images = self._prepare_X(images)
+        X_test, y_test = images, to_categorical(classes, num_classes = 10)
+        
+        # Compute test accuracy
+        (test_loss, test_acc) = self._model.evaluate(X_test, y_test)
+        return test_acc
+
+
+    def predict(self, queries):
+        X = self._prepare_X(queries)
+        probs = self._model.predict_proba(X)
+        return probs.tolist()
+
+
+    def destroy(self):
+        pass
+
+
+    def dump_parameters(self):
+        params = {}
+        # Save model parameters
+        model_bytes = pickle.dumps(self._model)
+        model_base64 = base64.b64encode(model_bytes).decode('utf-8')
+        params['model_base64'] = model_base64
+        return params
+
+
+    def load_parameters(self, params):
+        # Load model parameters
+        model_base64 = params.get('model_base64', None)
+        if model_base64 is None:
+                raise InvalidModelParamsException()
+                model_bytes = base64.b64decode(params['model_base64'].encode('utf-8'))
+                self._model = pickle.loads(model_bytes)
+
+
+    def _prepare_X(self, images):
+        X = np.asarray(images)
+        return X.reshape(-1,28,28,1)
+
+
+    def _build_classifier(self, epochs, batch_size):
+        model = models.Sequential()
+        model.add(Conv2D(filters = 32, kernel_size = (5,5),padding = 'Same', activation ='relu', input_shape = (28,28,1)))
+        model.add(Conv2D(filters = 32, kernel_size = (5,5),padding = 'Same', activation ='relu'))
+        model.add(layers.MaxPooling2D((2, 2)))
+        model.add(Dropout(0.25))
+        model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same', activation ='relu'))
+        model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same', activation ='relu'))
+        model.add(layers.MaxPooling2D((2, 2)))
+        model.add(Dropout(0.25))
+        model.add(Flatten())
+        model.add(Dense(256, activation = "relu"))
+        model.add(BatchNormalization())
+        model.add(Dense(10, activation = "softmax"))
+        my_optimizer = RMSprop(lr=0.001, rho=0.9, epsilon=1e-08, decay=0.0)
```

Comment:
For both models, can we add learning rate `lr` to the knob configuration? It is probably one of the most important hyperparameters (ref https://machinelearningmastery.com/understand-the-dynamics-of-learning-rate-on-deep-learning-neural-networks/)

Reply:
changed the code accordingly. set `lr` as the knob configuration to be tuned for both models. tested based on new documentation and get ~90% accuracy